app/services/RestaurantService.java

```diff
@@ -172,61 +172,44 @@ public Restaurant getRestaurantWithId(final UUID id)
      * @return the popular restaurants
      */
     @SuppressWarnings("unchecked")
-    public List<Restaurant> getPopularRestaurants(UUID userId)
+    public PaginationAdapter<Restaurant> getPopularRestaurants(User user)
     {
+	if (user != null)
+	{
 
-	UserHabitsBean userHabits = getUserHabits(userId);
-      
-	List<Restaurant> restaurants = (List<Restaurant>) getSession().createCriteria(Restaurant.class)
-		.add(Restrictions.eq("city.id", userHabits.getCity()))
-		.add(Restrictions.eq("priceRange", userHabits.getUserPriceRange())).list();
+	    UserHabitsBean userHabits = getUserHabits(user);
+	    if (userHabits != null)
+	    {
 
-	List<Object> toRemove = new ArrayList<Object>();
+		List<Restaurant> restaurants = (List<Restaurant>) getSession().createCriteria(Restaurant.class)
+			.add(Restrictions.eq("city.id", userHabits.getCity()))
+			.add(Restrictions.eq("priceRange", userHabits.getUserPriceRange())).list();
```

Comment:
There is still no limit on the query

Reply:
Kako nije restrikcija za rejting uradjena u upitu, nisam stavljao limit kako bi imao dovoljan broj restorana za ovo "rucno"izbacivanje